src/server.cpp

```diff
@@ -1577,7 +1577,10 @@ void Server::SendShowFormspecMessage(session_t peer_id, const std::string &forms
 	NetworkPacket pkt(TOCLIENT_SHOW_FORMSPEC, 0 , peer_id);
 	if (formspec.empty()){
 		//the client should close the formspec
-		m_formspec_state_data.erase(peer_id);
+		//but make sure there wasn't another one open in meantime
+		if (m_formspec_state_data[peer_id] == formname) {
```

Comment:
Use `m_formspec_state_data.find(peer_id)` instead. If this key/value pair did not exist, it's created by comparing it. Hence, f7a8e75 was required earlier to fix the empty string value.

Reply:
Wov. Another unexpected C++ feature. Fixing ...

Reply:
Will take time to check. C++ compiles extremely slowly ...

Reply:
... and it works. Here you go.